codegen/src/api/mod.rs

```diff
@@ -321,11 +321,14 @@ impl RuntimeGenerator {
             .collect();
         let pallet_names_len = pallet_names.len();
 
-        let metadata_hash = self
+        let runtime_api_names: Vec<_> = self
             .metadata
-            .hasher()
-            .only_these_pallets(&pallet_names)
```

Comment:
I removed the `only_these_pallets(&pallet_names)` here because I feld it is unnecessary. It includes all pallet_names from the metadata anyway.

Reply:
I think that makes sense now; we should make sure that the hashes actually do validate properly but I think this is all good :)